pkg/cloud/utils.go

```diff
@@ -233,21 +262,32 @@ func RelativeResourceName(project, resource string, key *meta.Key) string {
 }
 
 // SelfLink returns the self link URL for the given object.
-func SelfLink(ver meta.Version, project, resource string, key *meta.Key) string {
+func SelfLink(apiGroup meta.APIGroup, ver meta.Version, project, resource string, key *meta.Key) string {
 	var prefix string
+
+	switch apiGroup {
+	case meta.APIGroupNetworkServices:
+		prefix = networkServicesPrefix
+	default:
```

Comment:
Can we make this an explicit match to empty?

Reply:
This function does not currently return an error. Since we're already updating the function signature to include API Group we could theoretically also return an error, but I think that would make updating projects that depend on this especially painful. We could also do what we're doing elsewhere in this function and add something like `invalid-version` to the returned value, but that does not seem particularly helpful. Any preference here?

Reply:
invalid-xxx will be logged so we can potentially see what happened. Given this is very rare, we would still like it to be detectable.

Reply:
Updated, thanks!